projects/js-packages/publicize-components/src/components/generated-image-preview/index.js

```diff
@@ -0,0 +1,103 @@
+import { getRedirectUrl, ThemeProvider } from '@automattic/jetpack-components';
+import apiFetch from '@wordpress/api-fetch';
+import { Spinner } from '@wordpress/components';
+import { select, useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import classNames from 'classnames';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import useImageGeneratorConfig from '../../hooks/use-image-generator-config';
+import styles from './styles.module.scss';
+
+const getMediaSourceUrl = media => {
+	return media?.media_details?.sizes?.large?.source_url || media?.source_url;
+};
+
+/**
+ * Fetches the preview of the generated image based on the post info
+ *
+ * @returns {React.ReactNode} The generated image preview.
+ */
+export default function GeneratedImagePreview() {
+	const [ generatedImageUrl, setGeneratedImageUrl ] = useState( null );
+	const [ isLoading, setIsLoading ] = useState( true );
+
+	const { customText, imageType, imageId, template } = useImageGeneratorConfig();
+	const { getMedia } = select( 'core' );
+	const { title, featuredImage } = useSelect( _select => ( {
+		title: _select( editorStore ).getEditedPostAttribute( 'title' ),
+		featuredImage: _select( editorStore ).getEditedPostAttribute( 'featured_media' ),
+	} ) );
+
+	const imageTitle = useMemo( () => customText || title || 'Your Title', [ customText, title ] );
```

Comment:
Do we need `Your Title` here? Can it be an empty string? If not, should we make it translateable?

Reply:
As discussed, removed this and added a header for the preview.
p1679930151849559/1679928768.658069-slack-C02JJ910CNL